### PR TITLE
MM-35518 - Add CodeQL check

### PR DESCRIPTION
**Pull request by srkgupta**
### Summary
Added step to run automated CodeQL checks. Once this file is merged, the codeql checks (code scanning checks) will automatically be performed on following conditions:

- new pull requests to master branch
- scheduled to run every week once (it automatically pulls the latest codeql checks added, if any)

### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-35518

### CODE REVIEW

**Comment by agarciamontoro** [comment]
@srkgupta, is this still a work in progress?

**Comment by srkgupta** [comment]
FYI @crspeller, 

The CodeQL checks were previously reverted in this [PR](https://github.com/mattermost/mattermost-plugin-playbooks/pull/560). I am re-enabling them. The Security team will be notified of any new CodeQL alerts and will take appropriate actions, i.e. 
- raise an alert in the corresponding PR (or) 
- dismiss it as false positive/won't fix. 

As of now, there are no new alerts generated.

**Comment by srkgupta** [comment]
> @srkgupta, is this still a work in progress?

@agarciamontoro Sorry, I completely missed at re-looking this PR once again. Thanks for bringing this to my attention. It's complete and ready for review.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'javascript' ]
```
Comment (agarciamontoro) [inline .github/workflows/codeql-analysis.yml:35]: Does `javascript` cover typescript as well? Or should we add it there?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '43 14 * * 2'
```
Comment (agarciamontoro) [inline .github/workflows/codeql-analysis.yml:21]: Just curious: why this weird time?

**Review by agarciamontoro — COMMENTED** [review]
Looks good, but I'm not 100% sure of what this does. Could you add a bit more context on the goals of this PR, @srkgupta? Thanks!

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (srkgupta) [reply .github/workflows/codeql-analysis.yml:35]: CodeQL handles both javascript and typescript when the matrix language is defined as javascript.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (srkgupta) [reply .github/workflows/codeql-analysis.yml:21]: Thus is just to ensure that we dont end up running multiple CodeQL checks at the same time. Currently there is a limit on max number of parallel CodeQL checks from a single organization. The purpose is to run the CodeQL checks once a week.

**Review by agarciamontoro — APPROVED** [review]
Thank you for answering my questions, @srkgupta! Approving now, but I have one more:

If there's an alert in a PR, should we block merging until your team has taken a look? Should we ping you, or does the team receive an automatic notification?

**Comment by srkgupta** [comment]
@agarciamontoro as of now, if there are any alerts generated on a PR, it will not block the PR from being merged. Its always advised for the owner/reviewer to glance the alert and verify if the alert is valid and requires some code changes (or) if its handled through some other restrictions in place and is a false positive. 

The Security team will be anyways be notified of any new CodeQL alerts and will take appropriate actions, i.e.

- raise an alert in the corresponding PR as a comment. 
- dismiss it as false positive/won't fix.

Even if the PR gets merged with an open CodeQL alert (due to emergency fixes / by mistake), the security team will create a separate ticket and follow up and ensure that the issue is fixed if its a valid issue.

**Comment by agarciamontoro** [comment]
Makes sense. Thanks for the explanations, @srkgupta :)

**Review by crspeller — APPROVED** [review]

**Pull request merged**